pyro/poutine/runtime.py

```diff
@@ -368,6 +365,7 @@ def _fn(
             )
             # apply the stack and return its return value
             apply_stack(msg)
+            assert msg["value"] is not None
```

Comment:
Is this always correct? All tests have passed

Reply:
@eb8680 ?